src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -257,11 +260,22 @@ bool java_bytecode_languaget::typecheck(
     get_message_handler());
 
   // now typecheck all
-  if(java_bytecode_typecheck(
-       symbol_table, get_message_handler(), string_refinement_enabled))
-    return true;
+  auto res = java_bytecode_typecheck(
+       symbol_table, get_message_handler(), string_refinement_enabled);
 
-  return false;
+  generate_java_generic_typet new_symbol_generator(symbol_table, get_message_handler());
```

Comment:
Nit: more meaningful name for this please, perhaps `java_instantiated_generic_class_generator`? Maybe you can think of something better (And shorter!) - `new_symbol_generator doesn't tell me what kind of symbols are being generated, could always just use the class name.